src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs

```diff
@@ -166,32 +175,179 @@ public void CompilerDataLogger_Dispose_ShouldReadSarifV1()
+            BinaryAnalyzerContext context = CreateTestContext();
+            List<ITelemetry> sendItems = TestSetup(SarifPath, context, Sarif.SarifVersion.Current, out CompilerDataLogger compilerDataLogger, fileSystem.Object, true);
+            compilerDataLogger.WriteException(context, "testException");
+            sendItems.Count.Should().Be(0);
+            context.Dispose();
+        }
+
+        [Fact]
+        public void CompilerDataLogger_WriteException_ShouldLogExecutionExceptionFromBaseLineTestSarif()
+        {
+            string sarifLogPath = Path.Combine(PEBinaryTests.BaselineTestDataDirectory, ExpectedFolder, SampleSarifPath);
+            var fileSystem = new Mock<IFileSystem>();
+            BinaryAnalyzerContext context = CreateTestContext();
+            List<ITelemetry> sendItems = TestSetup(sarifLogPath, context, Sarif.SarifVersion.Current, out CompilerDataLogger compilerDataLogger, fileSystem.Object);
+            SarifLog sarifLog = SarifLog.Load(sarifLogPath);
+            ExecutionException exception = AnalysisSummaryExtractor.ExtractExceptionData(sarifLog).FirstOrDefault();
+            AnalysisSummary summary = AnalysisSummaryExtractor.ExtractAnalysisSummary(sarifLog, "", null);
+            compilerDataLogger.WriteException(exception, summary);
+            sendItems.Count.Should().Be(1);
+            context.Dispose();
+        }
+
+        [Fact]
+        public void CompilerDataLogger_WriteException_ShouldLogExecutionException()
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            BinaryAnalyzerContext context = CreateTestContext();
+            ExecutionException exception = new ExecutionException(type: "test", message: "testExecutionException", stackTrace: "testStackTrace", innerException: new Exception());
+            AnalysisSummary summary = new AnalysisSummary();
+            List<ITelemetry> sendItems = TestSetup(SarifPath, context, Sarif.SarifVersion.Current, out CompilerDataLogger compilerDataLogger, fileSystem.Object);
+            compilerDataLogger.WriteException(exception, summary);
+            sendItems.Count.Should().Be(1);
+            context.Dispose();
+        }
+
+        [Fact]
+        public void CompilerDataLogger_CreateScvOutputFile_ShouldSkipIfPathIsNull()
+        {
+            BinaryAnalyzerContext context = CreateTestContext();
+            var fileSystem = new Mock<IFileSystem>();
+            List<ITelemetry> sendItems = TestSetup(SarifPath, context, Sarif.SarifVersion.Current, out CompilerDataLogger compilerDataLogger, fileSystem.Object);
+            compilerDataLogger.CreateCsvOutputFile(null, false);
+            sendItems.Count.Should().Be(0);
+            context.Dispose();
+        }
+
+        private List<ITelemetry> TestSetup(string sarifLogFilePath, BinaryAnalyzerContext context, Sarif.SarifVersion sarifVersion, out CompilerDataLogger logger, IFileSystem fileSystem = null, bool isDisabledLogger = false)
```

Comment:
Look into reducing line length by dropping parameters to new lines via style guidelines.